#!/usr/bin/env python

import sys

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t

gen = ParameterGenerator()

gen.add(
    "std_warn_thresh_xy",
    double_t,
    0,
    "Warning threshold for the XY position uncertainty",
    sys.float_info.max,
    0,
    sys.float_info.max,
)
gen.add(
    "std_warn_thresh_z",
    double_t,
    0,
    "Warning threshold for position Z coordinate uncertainty",
    sys.float_info.max,
    0,
    sys.float_info.max,
)
gen.add(
    "std_warn_thresh_yaw",
    double_t,
    0,
    "Warning threshold for yaw orientation uncertainty",
    sys.float_info.max,
    0,
    sys.float_info.max,
)


PACKAGE = "mcl_3dl"
exit(gen.generate(PACKAGE, "mcl_3dl", "MCL3DLParams"))
